Python/pylifecycle.c

```diff
@@ -1257,6 +1257,20 @@ Py_FinalizeEx(void)
     PyThreadState *tstate = _PyRuntimeState_GetThreadState(runtime);
     PyInterpreterState *interp = tstate->interp;
 
+    // Finalize sub-interpreters.
+    runtime->interpreters.finalizing = 1;
+    PyInterpreterState *subinterp = PyInterpreterState_Head();
```

Comment:
Yeah I didn't like this when I wrote it - it was because `interp` is already used. I've changed to `curr_interp` (since we're looping over them and also store `next_interp`).